doc/build_your_own_container_image_howto.md

```diff
@@ -0,0 +1,71 @@
+# Build your own container image HOWTO
+
+This document describes how to build your own container image for the StarRocks Operator. There are some reasons why you
+might want to build your own container image, for example:
+
+1. You want to add fields to the StarRocksCluster CRD.
+2. Your k8s version is too old to support the latest StarRocks Operator image.
+
+This document will guide you through the process of building your own container image for the StarRocks Operator.
+
+## Prerequisites
+
+1. Automated build tool Make installed.
+2. Git installed.
+3. Go installed.
+4. [Docker](https://docs.docker.com/get-docker/) installed.
+5. A container image repository, such as [Docker Hub](https://hub.docker.com/).
+
+In order to contextualize this process, we assume that your k8s does not support appProtocol, and you want to remove it
+from the Operator.
+
+## 1. Find the commit ID you want to revert
```

Comment:
describe the standard process of building the image from the latest commit on main branch, and then describe how to tune the commits and build whatever wanted.

The main purpose is to build image from source code. And then it is just a matter of fact how to get the expected source code in shape, no matter it is a revert of some commits or make some additional changes to the code